File: go/274.go
package main

import "fmt"

func hIndex(citations []int) int {
	n := len(citations)
	cnt := make([]int, n+1)
	for _, t := range citations {
		if t > n {
			cnt[n]++
		} else {
			cnt[t]++
		}
	}
	fmt.Printf("cnt: %v\n", cnt)
	for i := n; i > 0; i-- {
		if cnt[i] >= i {
			return i
		} else {
			cnt[i-1] += cnt[i]
		}
	}
	return 0
}

func main() {

}

File: python/274.py
from typing import List

class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        cnt = [0] * (n + 1)
        for times in citations:
            if times > n:
                cnt[n] += 1
            else:
                cnt[times] += 1
        for i in range(n, 0, -1):
            if cnt[i] >= i:
                return i
            cnt[i-1] += cnt[i]
        return 0
        